Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -126,9 +114,27 @@ class GraphLinter: GraphLinting {
             issues.append(issue)
         }
 
+        issues.append(contentsOf: lintStaticDependencies(from: from, to: to))
+
         return issues
     }
 
+    private func lintStaticDependencies(from: TargetNode, to: TargetNode) -> [LintingIssue] {
+        guard to.target.product.isStatic, from.target.canLinkStaticProducts() else {
+            return []
+        }
+        let warning = StaticDepedencyWarning(targetNode: from,
```

Comment:
`targetNode` and `linkintStaticTargetNode` as names is misleading. I'd rather call them `fromTargetNode` and `toTargetNode`.